vpc/tool/identity/ec2.go

```diff
@@ -31,15 +35,63 @@ var (
 	getIdentitySuccess = stats.Int64("getIdentity.success.count", "How many times getIdentity succeeded", "")
 )
 
+const (
+	InstanceIdentityFile = "/run/instanceIdentity/doc"
+ 	PkcsFile = "/run/instanceIdentity/pkcs"
+)
+
 type ec2Provider struct{}
 
 func (e *ec2Provider) GetIdentity(ctx context.Context) (*vpcapi.InstanceIdentity, error) {
+	var pkcsFile *os.File
 	ctx, span := trace.StartSpan(ctx, "GetIdentity")
 	defer span.End()
 	start := time.Now()
 	stats.Record(ctx, getIdentityCount.M(1))
 
 	newAWSLogger := &awsLogger{logger: logger.G(ctx), oldMessages: list.New()}
+
+	// Attempt to read the data off of /run
+	instanceIdentityFile, err := os.OpenFile(InstanceIdentityFile, os.O_RDONLY, 0644)
+	if errors.Is(err, os.ErrNotExist) || err != nil { //should we separate out the does not exist with the regular error?
+		tracehelpers.SetStatus(ErrEC2MetadataServiceUnavailable, span)
+		goto fetchFromImds
+	}
+	pkcsFile, err = os.OpenFile(PkcsFile, os.O_RDONLY, 0644)
+	if errors.Is(err, os.ErrNotExist) || err != nil { //should we separate out the does not exist with the regular error?
+		// handle the case where the file doesn't exist
+		tracehelpers.SetStatus(ErrPkcsFileNotFound, span)
```

Comment:
If the file doesn't exist, why not try to perform the below action, and retry? Or launch a goroutine to write the files out?

Reply:
done.